code/scripting/api/objs/enums.cpp

```diff
@@ -206,6 +206,69 @@ const lua_enum_def_list Enumerations[] = {
 	{"COMMIT_MULTI_NOT_ALL_ASSIGNED", LE_COMMIT_MULTI_NOT_ALL_ASSIGNED, true},
 	{"COMMIT_MULTI_NO_PRIMARY", LE_COMMIT_MULTI_NO_PRIMARY, true},
 	{"COMMIT_MULTI_NO_SECONDARY", LE_COMMIT_MULTI_NO_SECONDARY, true},
+	{"SQUAD_MESSAGE_ATTACK_TARGET", LE_SQUAD_MESSAGE_ATTACK_TARGET, true},
+	{"SQUAD_MESSAGE_DISABLE_TARGET", LE_SQUAD_MESSAGE_DISABLE_TARGET, true},
+	{"SQUAD_MESSAGE_DISARM_TARGET", LE_SQUAD_MESSAGE_DISARM_TARGET, true},
+	{"SQUAD_MESSAGE_PROTECT_TARGET", LE_SQUAD_MESSAGE_PROTECT_TARGET, true},
+	{"SQUAD_MESSAGE_IGNORE_TARGET", LE_SQUAD_MESSAGE_IGNORE_TARGET, true},
+	{"SQUAD_MESSAGE_FORMATION", LE_SQUAD_MESSAGE_FORMATION, true},
+	{"SQUAD_MESSAGE_COVER_ME", LE_SQUAD_MESSAGE_COVER_ME, true},
+	{"SQUAD_MESSAGE_ENGAGE_ENEMY", LE_SQUAD_MESSAGE_ENGAGE_ENEMY, true},
+	{"SQUAD_MESSAGE_CAPTURE_TARGET", LE_SQUAD_MESSAGE_CAPTURE_TARGET, true},
+	{"SQUAD_MESSAGE_REARM_REPAIR_ME", LE_SQUAD_MESSAGE_REARM_REPAIR_ME, true},
+	{"SQUAD_MESSAGE_ABORT_REARM_REPAIR", LE_SQUAD_MESSAGE_ABORT_REARM_REPAIR, true},
+	{"SQUAD_MESSAGE_STAY_NEAR_ME", LE_SQUAD_MESSAGE_STAY_NEAR_ME, true},
+	{"SQUAD_MESSAGE_STAY_NEAR_TARGET", LE_SQUAD_MESSAGE_STAY_NEAR_TARGET, true},
+	{"SQUAD_MESSAGE_KEEP_SAFE_DIST", LE_SQUAD_MESSAGE_KEEP_SAFE_DIST, true},
+	{"SQUAD_MESSAGE_DEPART", LE_SQUAD_MESSAGE_DEPART, true},
+	{"SQUAD_MESSAGE_DISABLE_SUBSYSTEM", LE_SQUAD_MESSAGE_DISABLE_SUBSYSTEM, true},
+	{"SQUAD_MESSAGE_LUA_AI", LE_SQUAD_MESSAGE_LUA_AI, true},
+	{"BUILTIN_MESSAGE_ATTACK_TARGET", LE_BUILTIN_MESSAGE_ATTACK_TARGET, true},
```

Comment:
You could autogen this the same way as the other builtin message stuff, so that new types are automatically registered here:
```cpp
#define X(NAME, ...) {"BUILTIN_MESSAGE_" # NAME, LE_BUILTIN_MESSAGE_ ## NAME, true },
BUILTIN_MESSAGE_TYPES
#undef X
```

Reply:
The Lua enum list is long and terrible and needs a complete redesign, indeed. But I'd rather match the rest of the enums than try to fix it here by doing something different than all the other enums in the list. That and these are in an array for the Lua Enums to code to work with so I'm not sure your version of autogen will work in this instance. They'd have to be genned and then inserted.

Reply:
I concur with Mjn here, while getting Lua enums proper would be good, it's out of scope for this.
We ideally want a PR that properly refactors enums in a way that they no longer are the same type for totally different enums, but which also keeps script compatibility.
But that's definitely its own PR

Reply:
(I think you realized this, but just to be sure we're on the same page, the generation works by transforming the source code with preprocessor macros. This specific technique is called an X-macro, and it's one of the few applications of macros that (to my knowledge) can't be replaced by C++ templates.)

I'm fine with leaving cleaning up this list as an Indefinite Future Thing(TM). Could you add a comment to missionmessage.h saying that new standard builtin messages should be added here too?

Reply:
A comment was kinda already there. I expanded it.